packages/@aws-cdk/aws-codebuild/lib/source.ts

```diff
@@ -537,11 +548,22 @@ abstract class ThirdPartyGitSource extends GitSource {
     if (this.webhookTriggersBatchBuild) {
       project.enableBatchBuilds();
     }
+    const buildStatusConfig = {
+        targetUrl: this.buildStatusUrl
+    }
+    if (typeof superConfig["buildStatusContext"] !== "undefined") {
+        // Github/GitHubEnterprise
+        buildStatusConfig["context"] = superConfig["buildStatusContext"]
+    } else if (typeof superConfig["buildStatusName"] !== "undefined") {
+        // BitBucket
+        buildStatusConfig["context"] = superConfig["buildStatusName"]
+    }
```

Comment:
Since we need to handle `buildStatusConfig` separately in the two subclasses anyway, let's just remove all of this code.